### ISSUE TITLE
Comments sometimes not shown

### ISSUE THREAD

**Issue by dhow** [xxRockOnxx/obs-fb-live-comments#2]
Hi,

Thanks for creating this! I've been using this for a few days, sometimes it just works but most of the time new comments are not shown. 

When I stream I have 3 instance of the URL open: 1) OBS, 2) Chrome on PC, and 3) Firefox on an Android tablet. I use Firefox on Android with a custom CSS add-on to enlarge font size so I can see it from distance. 

After streaming I can see some comments are shown on some or non of the above browsers, sometimes all comments are shown. 

I'm not sure how to reproduce this but can help debug this issue. 

Thanks again!

**Comment by lemuelflores** [comment]
Hi,

Yes I am aware of this issue now that I'm using this again now more frequently especially since there's nothing to do during this time of pandemic but to play and share my stream.

Apparently, they're smart enough to not show comments from real users on a test app unless they are a test user as well. Replies to comments not from test users will not be shown.

I currently have an upcoming Major update which makes everything more simpler by simply just clicking a button to login with Facebook and copying a link but it requires an App Review from the Facebook team.

They require it so that they can check if your app is nothing suspicious or whatever their criteria is before they let you use their user's data.

Right now all I need to have to be able to submit my app for Review is an icon... which I never thought about. What I thought was a simple app that no one would actually use now is being used.

If you got any suggestions or can create an icon that would be awesome but I am actually not expecting someone to do it for me.

~~Once the app is approved, I will release it as soon as possible.~~

**Comment by dhow** [comment]
> Major update which makes everything more simpler by simply just clicking a button to login with Facebook and copying a link

I'd love to see that update -- simplify a lot of steps and reduce human error. 

> If you got any suggestions or can create an icon that would be awesome

Here you go. (-; 

![facebook comment app icon](https://user-images.githubusercontent.com/3618837/79453480-b873db80-8024-11ea-8b99-9eddf538205b.png)

So far I only stream with my personal account, but yesterday I created a Facebook page and I'll stream and let you know if I see any issue.

**Comment by lemuelflores** [comment]
App is now submitted for review. If their Review team isn't affected that much with the current pandemic, it should be reviewed within approximately 5 days from today.

This issue will be closed once everything is ok.

**Comment by dhow** [comment]
Cool -- great to know. Thanks for all your effort! (-;

**Comment by lemuelflores** [comment]
Hey, I got the response from them. It was pretty quick.

Unfortunately the app will not be getting approved as explained [here](https://github.com/xxRockOnxx/obs-fb-live-comments/blob/16698459d00ab5a957e7e55aafcf2c4dbda4b973/README.md).

I also apologize for updating and breaking the website frequently this past few days.

- The old instructions would work only on comments by the test user as explained earlier in this thread.
- Next, I made an app instead that simplifies all this thing by simply logging in and copying a link but unfortunately it wasn't approved so there's not point in pushing it forward.
- Now, I have updated the instructions by doing a bit of a hack that I'm pretty sure will work. 
  - Basically you just load the actual Facebook page in OBS
  - remove all the elements except for the comments box
  - make the background transparent
  - done!

Given that this method is just a hack, if Facebook updated their HTML code, there's a chance that this method will be broken too.

In that case, ping me if anyone experienced it before I do.

**Comment by JorgeNachtigall** [comment]
Oh, this sucks @xxRockOnxx ! I was creating an app similar to the streamyard. My mainly objective is retrieve all comments from my livestream using [https://developers.facebook.com/docs/graph-api/server-sent-events/endpoints/live-comments/](THIS) endpoint here (that uses SSE), but when I put this to work in my stream, I only can see the name and ID from a comment when I'm the author. For other users I only can see their comments. Any advices or should I just forget it?

EDIT: Yeah, i will need the user_videos permissions. This sucks so hard. I saw other day an implementation using regex to request the comments from livestream page, but, you know, chat is a real-time thing, and making a lot of request for facebook pages... will not work hehe

EDIT2: You think that, using maybe a method like what you have right now, I could extract the comments from the page "locally only"? Maybe identifying when a new div comment is added or something like this and sending this locally to a nodejs server. I want to display those comments in a "new visual chat" to add more functionalities, like: obs streamear clicks on a chat comment and then this appears in spotlight on the stream screen (I've already made all of this, but as I said above: only works for my user, for others I only can get their comments but no name and no user id) :(

**Comment by lemuelflores** [comment]
@JorgeNachtigall for personal facebook streams unfortunately this is our only way right now. the only problem is it's a bit unreliable because it relies heavily on DOM which is very susceptible to changes. The only way for me to catch up with the changes is if someone notifies me but it seems like not everyone can be bothered.

For pages I think you can create an app for that. I think https://streamelements.com/ does that. Not sure cause I haven't tried it.
